lib/battle/BattleHex.cpp

```diff
@@ -145,28 +146,31 @@ std::vector<BattleHex> BattleHex::allNeighbouringTiles() const
 	std::vector<BattleHex> ret;
 	ret.resize(6);
 
-	for(EDir dir = EDir(0); dir <= EDir(5); dir = EDir(dir+1))
+	for(EDir dir = static_cast<EDir>(0); dir <= static_cast<EDir>(5); dir = static_cast<EDir>(dir + 1))
 		ret[dir] = cloneInDirection(dir, false);
 
 	return ret;
 }
 
 BattleHex::EDir BattleHex::mutualPosition(BattleHex hex1, BattleHex hex2)
 {
-	for(EDir dir = EDir(0); dir <= EDir(5); dir = EDir(dir+1))
-		if(hex2 == hex1.cloneInDirection(dir,false))
+	for(EDir dir = static_cast<EDir>(0); dir <= static_cast<EDir>(5); dir = static_cast<EDir>(dir + 1))
+		if(hex2 == hex1.cloneInDirection(dir, false))
 			return dir;
 	return NONE;
 }
 
 uint8_t BattleHex::getDistance(BattleHex hex1, BattleHex hex2)
 {
-	int y1 = hex1.getY(), y2 = hex2.getY();
+	int y1 = hex1.getY();
+	int y2 = hex2.getY();
 
 	// FIXME: Omit floating point arithmetics
-	int x1 = (int)(hex1.getX() + y1 * 0.5), x2 = (int)(hex2.getX() + y2 * 0.5);
+	int x1 = static_cast<int>(hex1.getX() + y1 * 0.5);

```

Comment:
@nullkiller maybe implicit conversion to float is significant, because y1 and y2 can be odd.

Reply:
@IvanSavenko what do you think, is division by 2 will change inner logic or not? I think there is no change, but I am cautious.

Reply:
I think this won't change anything.
However I don't understand:
- why this FIXME was added to begin with. It implies that this is not a trivial change
- what exactly this code is supposed to do?

I think this method needs some investigation - check how it used, check how it is *supposed* to work, check how it actually works, and finally - clarify this code.

I guess you could remove float conversion, but it won't improve readability here much. This is still an obscure code that should be looked into, either now or at some point later.

Reply:
@IvanSavenko agree. But I think than later is better, and you already wrote a reasons to @nullkiller .

Reply:
changed to /2

Reply:
/2 logically equivalent as getX() gives int and adding int to float will not change decimal part and we can have only 0 or 0.5 as decimal part so when casting to int it will be always ignored. Looks like this methods tries to calculate some distance between two points taking into account that the field is hexagonal.